eth/beacon/state_machines/base.py

```diff
@@ -193,9 +222,562 @@ def active_state(self) -> ActiveState:
+                "Attestation slot number too high:\n"
+                "\tFound: %s Needed less than or equal to %s" %
+                (attestation.slot, parent_block.slot_number)
+            )
+        if attestation.slot < max(parent_block.slot_number - cycle_length + 1, 0):
+            raise ValidationError(
+                "Attestation slot number too low:\n"
+                "\tFound: %s, Needed greater than or equalt to: %s" %
+                (
+                    attestation.slot,
+                    max(parent_block.slot_number - cycle_length + 1, 0)
+                )
+            )
+
+    @classmethod
+    def validate_justified(cls,
+                           crystallized_state: CrystallizedState,
+                           attestation: 'AttestationRecord',
+                           chaindb: BaseBeaconChainDB) -> None:
+        """
+        Validate ``justified_slot`` and ``justified_block_hash`` fields.
+
+        Raise ``ValidationError`` if it's invalid.
+        """
+        if attestation.justified_slot > crystallized_state.last_justified_slot:
+            raise ValidationError(
+                "attestation.justified_slot %s should be equal to or earlier than"
+                " crystallized_state.last_justified_slot %s" % (
+                    attestation.justified_slot,
+                    crystallized_state.last_justified_slot,
+                )
+            )
+
+        justified_block = chaindb.get_block_by_hash(attestation.justified_block_hash)
+        if justified_block is None:
+            raise ValidationError(
+                "justified_block_hash %s is not in the canonical chain" %
+                attestation.justified_block_hash
+            )
+        if justified_block.slot_number != attestation.justified_slot:
+            raise ValidationError(
+                "justified_slot %s doesn't match justified_block_hash" % attestation.justified_slot
+            )
+
+    @classmethod
+    def validate_bitfield(cls,
+                          attestation: 'AttestationRecord',
+                          attestation_indices: Sequence[int]) -> None:
+        """
+        Validate ``attester_bitfield`` field.
+
+        Raise ``ValidationError`` if it's invalid.
+        """
+        if len(attestation.attester_bitfield) != get_bitfield_length(len(attestation_indices)):
+            raise ValidationError(
+                "Attestation has incorrect bitfield length. Found: %s, Expected: %s" %
+                (len(attestation.attester_bitfield), get_bitfield_length(len(attestation_indices)))
+            )
+
+        # check if end bits are zero
+        last_bit = len(attestation_indices)
+        if last_bit % 8 != 0:
+            for i in range(8 - last_bit % 8):
+                if has_voted(attestation.attester_bitfield, last_bit + i):
+                    raise ValidationError("Attestation has non-zero trailing bits")
+
+    @classmethod
+    def validate_aggregate_sig(cls,
```

Comment:
same about `staticmethod` -> extraction to standalone (since `create_signing_message` is also able to be made standalone.